index.d.ts

```diff
@@ -1,184 +1,261 @@
-import { Dispatch, SetStateAction, ReactNode } from 'react';
-
 declare module 'react-table' {
-    export interface Cell<D> {
-        render: (type: string) => any;
-        getCellProps: () => any;
-        column: Column<D>;
-        row: Row<D>;
-        state: any;
-        value: any;
-    }
+  import { ReactNode, ReactText, Dispatch, SetStateAction } from 'react'
+    export type SortingRule<D> = {
+    id: keyof D
+    desc: boolean
+  }
 
-    export interface Row<D> {
-        index: number;
-        cells: Cell<D>[];
-        getRowProps: () => any;
-        original: any;
-    }
+  export type SortingRules<D> = SortingRule<D>[]
 
-    export interface HeaderColumn<D, A extends keyof D = never> {
-        /**
-         * This string/function is used to build the data model for your column.
-         */
-        accessor: A | ((originalRow: D) => string);
-        Header?: string | ((props: TableInstance<D>) => ReactNode);
-        Filter?: string | ((props: TableInstance<D>) => ReactNode);
-        Cell?: string | ((cell: Cell<D>) => ReactNode);
-
-        /**
-         * This is the unique ID for the column. It is used by reference in things like sorting, grouping, filtering etc.
-         */
-        id?: string | number;
-        minWidth?: string | number;
-        maxWidth?: string | number;
-        width?: string | number;
-        canSortBy?: boolean;
-        sortByFn?: (a: any, b: any, desc: boolean) => 0 | 1 | -1;
-        defaultSortDesc?: boolean;
-    }
+  export type Filters<D> = {
+    [key: keyof D]: string
+  }
 
-    export interface Column<D, A extends keyof D = never> extends HeaderColumn<D, A> {
-        id: string | number;
-    }
+  export interface Cell<D = any> extends TableInstance<D> {
```

Comment:
Should we make D extend an object? since we are looking for keys? This feels like a bit too broad of a type based on what we know.

Reply:
makes sense.

Reply:
it also causes a lot of lazy typing around unspecified generics to show up - pretty easily fixed in my code.  So all in all I suspect that this is a good thing.